File: web-server/src/main/java/com/churchofphilippi/webserver/model/Notification.java
package com.churchofphilippi.webserver.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;

@Entity
@Table(name = "notification")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Notification {

    @Id
    @SequenceGenerator(
            name = "notification_sequence",
            sequenceName = "notification_sequence",
            allocationSize = 1
    )
    @GeneratedValue(
            strategy = GenerationType.SEQUENCE,
            generator = "notification_sequence"
    )
    @Column(
            name = "notification_id",
            updatable = false
    )
    private Long notificationId;

    @Column(
            nullable = false,
            columnDefinition = "TEXT"
    )
    private String content;

    @Column(nullable = false)
    private LocalDateTime creationDate;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dept_id")
    @JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
    private Dept department;

}

File: cop_mobile/cop_mobile/cop_mobile/Model/Notification.cs
using cop_mobile.Services;
using System;
using System.Collections.Generic;
using System.Text;

namespace cop_mobile.Model
{
    public class Notification
    {
        public long notificationId { get; set; }
        public string content { get; set; }
        public DateTime creationDate { get; set; }
        public Department department { get; set; }
        public Boolean isChurchWide => this.department != null;
        public string title;
        public string relativeTime => BaseService.getRelativeTime(creationDate);

        public Notification(long notificationId, string content, DateTime creationDate, Department department)
        {
            this.notificationId = notificationId;
            this.content = content;
            this.creationDate = creationDate;
            this.department = department;
        }

        public string Title
        {
            get
            {
                if (this.department != null)
                {
                    return this.department.name;
                }
                else
                {
                    return "";
                }
            }
        }
    }
}
